elide-core/src/main/java/com/yahoo/elide/jsonapi/document/processors/WithMetadata.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.jsonapi.document.processors;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * The class carries metadata fields.
+ */
+public interface WithMetadata {
+
+    /**
+     * Sets a metadata property for this request.
+     * @param property
+     * @param value
+     */
+    void setMetadataField(String property, Object value);
```

Comment:
Is this required? Doesn't look like it is used internally? Is it just in the case for a custom datastore? Should we add a `noop` default interface implementation?